feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/ui/PreviewScreenComponents.kt

```diff
@@ -471,37 +472,56 @@ fun PreviewDisplay(
 
 @Composable
 fun StabilizationIcon(
-    stabilizationMode: StabilizationMode,
-    modifier: Modifier = Modifier,
-    active: Boolean = true
+    stabilizationUiState: StabilizationUiState.Set,
+    modifier: Modifier = Modifier
 ) {
     val contentColor = Color.White.let {
-        if (!active) it.copy(alpha = 0.38f) else it
+        if (!stabilizationUiState.active) it.copy(alpha = 0.38f) else it
     }
     CompositionLocalProvider(LocalContentColor provides contentColor) {
-        if (stabilizationMode != StabilizationMode.OFF) {
+        if (stabilizationUiState.stabilizationMode != StabilizationMode.OFF) {
             Icon(
-                painter = when (stabilizationMode) {
-                    StabilizationMode.AUTO ->
-                        painterResource(R.drawable.video_stable_auto_filled_icon)
-
-                    StabilizationMode.HIGH_QUALITY ->
-                        painterResource(R.drawable.video_stable_hq_filled_icon)
-
-                    else -> rememberVectorPainter(Icons.Filled.VideoStable)
+                painter = when (stabilizationUiState) {
+                    is StabilizationUiState.Specific ->
+                        when (stabilizationUiState.stabilizationMode) {
+                            StabilizationMode.AUTO ->
```

Comment:
I think I'm getting a bit mixed up here. When would we have `StabilizaitonUiState.Specific` with a `stabilizationMode` value of `AUTO`, and when would we have a `StabilizationUiState.AUTO`?

Reply:
Good catch. I missed removing that when I added the `Auto` subclass. It now throws an `IllegalStateException` since we should never have `AUTO` as a `Specific` type.